### PR TITLE
Add About Dialog

### PR DESCRIPTION
**Pull request by JakobDev**
I added just a little about dialog. You may want to modify Config.hpp-in so it takes the tag name (without the leading v) instead of changing it per hand every update.

grafik

### CODE REVIEW

File: Source/RMG/UserInterface/Dialog/AboutDialog.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ *  along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "AboutDialog.hpp"
+
+#include "Config.hpp"
+
+using namespace UserInterface::Dialog;
+
+AboutDialog::AboutDialog()
+{
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.cpp:17]: This should accept have an argument `QObject* parent` and pass it through the `QDialog` like
```cpp
AboutDialog::AboutDialog(QObject* parent) : QDialog(parent, Qt::WindowSystemMenuHint | Qt::WindowTitleHint)
{
}
```

File: Source/RMG/UserInterface/Dialog/AboutDialog.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    this->setupUi(this);
+
+    this->versionLabel->setText(QString("RMG version %1").arg(VERSION));
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.cpp:20]: Please change this to `RMG %1`

File: Source/RMG/UserInterface/Dialog/AboutDialog.ui [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     </property>
+     <property name="wordWrap">
+      <bool>true</bool>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <widget class="QLabel" name="label_3">
+     <property name="text">
+      <string>This Program is licensed under GPL 3</string>
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.ui:78]: Please change `GPL 3` to `GPLv3`

File: Source/RMG/UserInterface/Dialog/AboutDialog.ui [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+     </property>
+     <property name="wordWrap">
+      <bool>true</bool>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <widget class="QLabel" name="label_4">
+     <property name="text">
+      <string>Copyright © 2020-2022 Rosalie241</string>
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.ui:91]: Please change `Rosalie241`  to `Rosalie Wanders`

File: Source/RMG/UserInterface/MainWindow.cpp [hunk 5]
```diff
@@ -1106,6 +1108,7 @@ void MainWindow::on_Action_Help_HomePage(void)
 
 void MainWindow::on_Action_Help_About(void)
 {
+    this->aboutDialog->exec();
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/MainWindow.cpp:1111]: Please don't put `aboutDialog` in the class members, just do
```cpp
Dialog::AboutDialog dialog(this);
dialog.exec();
```

And please remove the include from `MainWindow.hpp` and put it in `MainWindow.cpp`

File: Source/RMG/UserInterface/Dialog/AboutDialog.ui [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        <set>Qt::AlignCenter</set>
+       </property>
+      </widget>
+     </item>
+    </layout>
+   </item>
+   <item>
+    <widget class="QLabel" name="versionLabel">
+     <property name="text">
+      <string notr="true">RMG version {version}</string>
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.ui:55]: Please change `version {version}` to `{version}`

File: Source/RMG/UserInterface/Dialog/AboutDialog.ui [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+     <property name="text">
+      <string>Copyright © 2020-2022 Rosalie241</string>
+     </property>
+     <property name="alignment">
+      <set>Qt::AlignCenter</set>
+     </property>
+    </widget>
+   </item>
+   <item>
+    <widget class="QDialogButtonBox" name="okButton">
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.ui:99]: This is a `ButtonBox` so just name it `buttonBox`

File: Source/RMG/Config.hpp.in [hunk 8]
```diff
@@ -4,6 +4,7 @@
 #define WINDOW_TITLE "Rosalie's Mupen GUI (${GIT_VERSION})"
 #define WINDOW_WIDGET_SIZE_W 640
 #define WINDOW_WIDGET_SIZE_H 480
+#define VERSION "0.1.1"
```
Comment (Rosalie241) [inline Source/RMG/Config.hpp.in:7]: Please change `VERSION` to `VERSION_STR` and change `0.1.1` to `${GIT_VERSION}`

File: Source/RMG/CMakeLists.txt [hunk 9]
```diff
@@ -22,6 +22,8 @@ set(RMG_SOURCES
     UserInterface/Dialog/SettingsDialog.qrc
     UserInterface/Dialog/RomInfoDialog.cpp
     UserInterface/Dialog/RomInfoDialog.ui
+    UserInterface/Dialog/AboutDialog.ui
+    UserInterface/Dialog/AboutDialog.cpp
```
Comment (Rosalie241) [inline Source/RMG/CMakeLists.txt:26]: Please swap these around like with `RomInfoDialog`, the `cpp` file first and `ui` file second.

File: Source/RMG/UserInterface/Dialog/AboutDialog.cpp [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+void AboutDialog::on_okButton_clicked(void)
```
Comment (Rosalie241) [inline Source/RMG/UserInterface/Dialog/AboutDialog.cpp:23]: Thos function is unneeded, please remove it.

**Review by Rosalie241 — CHANGES_REQUESTED** [review]
Thank you for the PR, please make the requested changes and I'll merge it :heart:

**Comment by Rosalie241** [comment]
> The requested changes are now done. I also disabled resizing of the About Window.

I'd prefer to keep resizing an option, so please revert that.

> But to be honest, I think showing the latest commit in the About Dialog is a bad idea. Is a version released as version 0.1.1, this is what should be shown in the About Dialog rather than the latest commit.

I'll change `GIT_VERSION` to use `git describe --tags --always` now, so on version `v0.1.2` it'll be `v0.1.2` then

**Comment by JakobDev** [comment]
Resizing is now enabled again, but it looks ugly when maximized.

**Comment by Rosalie241** [comment]
Thank you very much :heart:

**Pull request merged**